Author: Sune Vuorela <sune@debian.org>
Author: Ralf Jung <ralfjung-e@gmx.de>
Description: Put protocols' Client classes in a namespace.
 This patch avoids crashes when terminating kopete, fixes these bugs:
Bug: https://bugs.kde.org/show_bug.cgi?id=268056
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=644197
Applied-Upstream: 4.8, commit:1257454 (for the kopete/protocols/oscar/* parts)

--- a/kopete/protocols/oscar/oscarprivacyengine.h
+++ b/kopete/protocols/oscar/oscarprivacyengine.h
@@ -33,7 +33,9 @@ class QComboBox;
 class QAbstractItemView;
 
 class OscarAccount;
+namespace Oscar {
 class Client;
+}
 
 class OSCAR_EXPORT OscarPrivacyEngine : public QObject
 {
@@ -68,7 +70,7 @@ private:
 	QStandardItemModel m_contactsModel;
 	QStandardItemModel m_allContactsModel;
 	
-	Client* m_client;
+	Oscar::Client* m_client;
 	Type m_type;
 	
 	QComboBox* m_comboBox;
--- a/kopete/protocols/oscar/liboscar/client.cpp
+++ b/kopete/protocols/oscar/liboscar/client.cpp
@@ -95,6 +95,8 @@ namespace
 	DefaultCodecProvider defaultCodecProvider;
 }
 
+namespace Oscar {
+
 class Client::ClientPrivate
 {
 public:
@@ -1832,5 +1834,7 @@ void Client::gotFileMessage( int type, c
 	kDebug(14151) << "nobody wants it :(";
 }
 
+}
+
 #include "client.moc"
 //kate: tab-width 4; indent-mode csands; space-indent off; replace-tabs off;
--- a/kopete/protocols/oscar/liboscar/client.h
+++ b/kopete/protocols/oscar/liboscar/client.h
@@ -55,7 +55,6 @@ namespace Kopete
 namespace Oscar
 {
 class Settings;
-}
 
 class LIBOSCAR_EXPORT Client : public QObject
 {
@@ -649,7 +648,8 @@ private:
 	class ClientPrivate;
 	ClientPrivate* d;
 };
-Q_DECLARE_OPERATORS_FOR_FLAGS(Client::ICQStatus)
+}
+Q_DECLARE_OPERATORS_FOR_FLAGS(Oscar::Client::ICQStatus)
 
 #endif
 
--- a/kopete/protocols/oscar/liboscar/connection.h
+++ b/kopete/protocols/oscar/liboscar/connection.h
@@ -25,7 +25,6 @@ Kopete (c) 2002-2005 by the Kopete devel
 #include "liboscar_export.h"
 
 class ConnectionPrivate;
-class Client;
 class ClientStream;
 class Transfer;
 class RateClassManager;
@@ -36,6 +35,7 @@ class QHostAddress;
 
 namespace Oscar
 {
+class Client;
 class Settings;
 
 class MessageInfo
@@ -68,7 +68,7 @@ public:
 
 	static void setStartFlapSequenceList( const QList<Oscar::WORD>& seqList );
 	
-	void setClient( Client* );
+	void setClient( Oscar::Client* );
 
 	void connectToServer( const QString& host, quint16 port );
 	/**
--- a/kopete/protocols/oscar/aim/aimchatsession.cpp
+++ b/kopete/protocols/oscar/aim/aimchatsession.cpp
@@ -25,6 +25,8 @@
 #include "kopeteprotocol.h"
 #include "client.h"
 
+using namespace Oscar;
+
 AIMChatSession::AIMChatSession( const Kopete::Contact* user,  Kopete::ContactPtrList others,
                                 Kopete::Protocol* protocol, Oscar::WORD exchange,
                                 const QString& room )
--- a/kopete/protocols/oscar/aim/aimchatsession.h
+++ b/kopete/protocols/oscar/aim/aimchatsession.h
@@ -22,7 +22,9 @@
 #include "kopetechatsession.h"
 #include "oscartypes.h"
 
+namespace Oscar {
 class Client;
+}
 
 class AIMChatSession : public Kopete::ChatSession
 {
@@ -37,7 +39,7 @@ public:
      * Set the engine to use so that we can disconnect from the chat service
      * properly
      */
-    void setEngine( Client* engine );
+    void setEngine( Oscar::Client* engine );
 
     /**
      * Get the name of the AIM chat room represented by
@@ -77,7 +79,7 @@ public:
 private:
     QString m_roomName;
     Oscar::WORD m_exchange;
-    Client* m_engine;
+    Oscar::Client* m_engine;
 };
 
 
--- a/kopete/protocols/oscar/oscaraccount.h
+++ b/kopete/protocols/oscar/oscaraccount.h
@@ -36,7 +36,9 @@ class Group;
 class FileTransferInfo;
 }
 
+namespace Oscar {
 class Client;
+}
 class ClientStream;
 class OscarContact;
 class OscarAccountPrivate;
@@ -52,7 +54,7 @@ public:
 	virtual ~OscarAccount();
 
 	/** Provide the derived accounts and contacts with access to the backend */
-	Client* engine();
+	Oscar::Client* engine();
 
 	/** Disconnects this account */
 	virtual void disconnect();
--- a/kopete/protocols/oscar/oscaraccount.cpp
+++ b/kopete/protocols/oscar/oscaraccount.cpp
@@ -73,6 +73,8 @@
 #include "oscarstatusmanager.h"
 #include <kopetesockettimeoutwatcher.h>
 
+using namespace Oscar;
+
 
 class OscarAccountPrivate : public Client::CodecProvider
 {
